include/circt/Dialect/MSFT/MSFTOps.td

```diff
@@ -133,7 +139,10 @@ def MSFTModuleOp : MSFTModuleOpBase<"module",
       TypeAttrOf<FunctionType>:$function_type,
       OptionalAttr<DictArrayAttr>:$arg_attrs,
       OptionalAttr<DictArrayAttr>:$res_attrs,
-      StrArrayAttr:$argNames, StrArrayAttr:$resultNames,
+      StrArrayAttr:$argNames,
+      StrArrayAttr:$resultNames,
+      LocationArrayAttr:$argLocs,
+      LocationArrayAttr:$resultLocs,
```

Comment:
@teqdruid This required a bunch of changes to the ESI / MSFT modules, are you ok with adding these attributes to MSFTModules?